docs/sdks/server-sdks/dotnet.md

```diff
@@ -49,60 +83,339 @@ internal class AssignmentLogger : IAssignmentLogger
+{
+    ["age"] = 30,
+    ["country"] = "uk",
+    ["pricingTier"] = "1"  // NOTE: Deliberately setting to string causes this to be treated as a categorical attribute
+};
 
-![Toggle to enable environment](/img/feature-flagging/enable-environment.png)
+myUserAttributes["last30DaySpend"] = userService.GetRecentDaysSpend(30);
+myUserAttributes.Add("profileCompletion", 0.50f);
+```
 
-3.  If `getStringAssignment` is invoked before the SDK has finished initializing, the SDK may not have access to the most recent experiment configurations. In this case, the SDK will assign a variation based on any previously downloaded experiment configurations stored in local storage, or return `null` if no configurations have been downloaded.
 
-<br />
 
-:::note
-It may take up to 10 seconds for changes to Eppo experiments to be reflected by the SDK assignments.
-:::
+## Full Initialization and Assignment Example
+
+```cs
+class Program
+{
+    public void main()
+    {
+
+        // Initialize Segment and Eppo clients.
+        var segmentConfig = new Configuration(
+                    "<YOUR WRITE KEY>",
+                    flushAt: 20,
+                    flushInterval: 30);
+        var analytics = new Analytics(segmentConfig);
+
+        // Create a logger to send data back to the Segment data warehouse
+        var logger = new SegmentLogger(analytics);
+
+        // Initialize the Eppo Client
+        var eppoClientConfig = new EppoClientConfig("EPPO-SDK-KEY-FROM-DASHBOARD", logger);
+        var eppoClient = EppoClient.Init(eppoClientConfig);
+
+        // Elsewhere in your code, typically just after the user logs in.
+        var subjectTraits = new JsonObject()
+        {
+            ["email"] = "janedoe@liamg.com",
+            ["age"] = 35,
+            ["accountAge"] = 2,
+            ["tier"] = "gold"
+        }; // User properties will come from your database/login service etc.
+        var userID = "user-123";
+
+        // Identify the user in Segment Analytics.
+        analytics.Identify(userID, subjectTraits);
+
+
+        // Need to reformat user attributes a bit; EppoClient requires `IDictionary<string, object?>`
+        var subjectAttributes = subjectTraits.Keys.ToDictionary(key => key, key => (object)subjectTraits[key]);
+
+        // Get assignments
+        var showUpgradeAd = eppoClient.GetBooleanAssignment(
+            "show-upgrade-ad",
+            userID,
+            subjectAttributes,
+            false
+        );
+        // Get an assignment for the user
+        var recentUserTip = eppoClient.GetStringAssignment(
+            "recent-user-onboarding",
+            userID,
+            subjectAttributes,
+            "control"
+        );
+
+        // Get a bandit action
+        var actions = new List<String>(){"nike", "adidas"};
+        var banditResult = eppoClient.GetBanditAction(
+          "shoe-bandit",
+          ContextAttributes.FromDictionary(userID, subjectAttributes),
```

Comment:
Any reason you're passing this instead of just `subjectAttributes`?

Reply:
That's actually one overload that's not available (subjectKey, subject Attributes and List<String> actions).